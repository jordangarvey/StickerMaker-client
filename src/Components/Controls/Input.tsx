import React, { FC } from "react";
import { TextInput, StyleSheet } from "react-native";


/**
 * Interface for the Input props
 */
interface IInputProps {
	/** Callback to be fired whenever the text value is changed */
	onChange(text: string): void;
	/** The currently selected text */
	value: string;

	/** Should this input auto focus? */
	autoFocus?: boolean;
	/** Optional placeholder text */
	placeholder?: string;
	/** Optional width for this component */
	width?: number | string;
}

/**
 * Component to render the Input component
 */
const Input: FC<IInputProps> = (props) => (
	<TextInput
		autoFocus={false}
		onChangeText={props.onChange}
		placeholder={props.placeholder}
		returnKeyType="done"
		style={[styles.input, { width: props.width || 200 }]}
		textContentType="none"
		value={props.value}
	/>
);

const styles = StyleSheet.create({
	input: {
		borderColor: "gray",
		borderWidth: 1, 
		fontSize: 14,
		height: 40,
		padding: 5
	}
});

export default Input;
