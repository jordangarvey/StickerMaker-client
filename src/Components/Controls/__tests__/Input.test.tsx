import React from "react";

import renderer from "react-test-renderer";

import Input from "../Input";


describe("The Inputcomponentâ€™s snapshot", () => {
	it("renders as expected", () => {
		function mockFunction() {
			console.log("Test")
		}

		const component = renderer.create(
			<Input
				onChange={mockFunction}
				value="Test"
			/>
		).toJSON();


		expect(component).toMatchSnapshot();
	});
});
