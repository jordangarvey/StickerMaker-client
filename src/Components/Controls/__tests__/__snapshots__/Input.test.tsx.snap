// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Inputcomponentâ€™s snapshot renders as expected 1`] = `
<TextInput
  allowFontScaling={true}
  autoFocus={false}
  onChangeText={[Function]}
  rejectResponderTermination={true}
  returnKeyType="done"
  style={
    Array [
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "fontSize": 14,
        "height": 40,
        "padding": 5,
      },
      Object {
        "width": 200,
      },
    ]
  }
  textContentType="none"
  underlineColorAndroid="transparent"
  value="Test"
/>
`;
